compiler/gentree.cpp

```diff
@@ -2051,16 +2056,16 @@ VertexPtr GenTree::get_statement(vk::string_view phpdoc_str) {
           || (next == tok_static && cur_function->type == FunctionData::func_class_holder)) {
         vk::string_view phpdoc_str = cur->str_val;
         next_cur();
-        return get_statement(phpdoc_str);
+        return get_statement(new PhpDocComment(phpdoc_str));
```

Comment:
Why not to use reliable semantic with std::unique_ptr or std::shared_ptr? (It's about a rule - don't use new/delete in new c++ code)

Reply:
It's not unique_ptr, because a single phpdoc could be kept in multiple places.
More semantically, it's shared, but like in any other place in our code, I don't see any reason for complification. We don't need to handle refcounts, we don't need to call destructors, all our objects live till compilation end and never freed. Keeping `const T *` in `FunctionData`, `ClassData`, `vertex_inner`, etc. is all over the code.